File: rb/features.rb
# PhishIn SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module PhishInFeatures
  def self.make_feature(name)
    case name
    when "base"
      PhishInBaseFeature.new
    when "test"
      PhishInTestFeature.new
    else
      PhishInBaseFeature.new
    end
  end
end

File: py/phishin_sdk/features.py
# PhishIn SDK feature factory

from phishin_sdk.feature.base_feature import PhishInBaseFeature
from phishin_sdk.feature.test_feature import PhishInTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PhishInBaseFeature(),
        "test": lambda: PhishInTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
